script/grblcon.py

```diff
@@ -603,6 +651,18 @@ def _open_port(self, portstr='/dev/ttyUSB0', baud=38400):
                 self._serial_thread.start()
             except (serial.serialutil.SerialException, FileNotFoundError):
                 self._log_crit("Unable to open port: {}".format(portstr))
+    
+    def poll_adcs(self,*args):
+        """Periodically polls the ADC values in grbl
+        
+        :returns: None
+        :rtype: None        
+
+        """
+        if self.connected:
+          self._write('|')
+        
+        self.loop.set_alarm_in(self.adc_poll_time, self.poll_adcs)
```

Comment:
No need for thread safe writes here as the writes are guaranteed to be single producer due to the way urwid works.  The polling of the serial port works the same way that the serial channel does (though it could use the daemon property set to makes sure it dies cleanly)